scvi/model/_scvi.py

```diff
@@ -280,7 +280,18 @@ def to_latent_mode(
             Key to use in `adata.obsm` where the latent qzm params are stored
         use_latent_qzv_key
             Key to use in `adata.obsm` where the latent qzv params are stored
+        attach_reduced_data
+            If `True`, sets `.adata` attribute with a memory-reduced anndata
+            in which all count data is set to zeros
         """
-        self._get_latent_adata_from_adata(mode, use_latent_qzm_key, use_latent_qzv_key)
-        self.adata_manager.register_new_fields(self.__class__._get_latent_fields(mode))
+        bdata = self._get_reduced_adata(mode)
+        if mode == _SCVI_LATENT_MODE:
+            bdata.obsm[_SCVI_LATENT_QZM] = self.adata.obsm[use_latent_qzm_key]
+            bdata.obsm[_SCVI_LATENT_QZV] = self.adata.obsm[use_latent_qzv_key]
+        else:
+            raise ValueError(f"Unknown latent mode: {mode}")
+        new_manager = self.adata_manager.transfer_fields(bdata)
+        new_manager.register_new_fields(self._get_latent_fields(mode))
+        self._adata_manager = new_manager
+        self.adata = bdata
```

Comment:
I am a bit confused. Aren't we overwriting the user's data by doing this, and if so then what is the point in making any of these changes? More precisely:

- If `id(self.adata) == id(user's adata)` then this is essentially overwriting the user's data _and_ the model's adata, so there is no point in making a copy since we are losing the original dataset regardless.

- If` id(self.adata) != id(user's adata) `then the user's data is anyways not altered with this call the way that it is today in the main branch, so again there is no point in making a copy.

🤔

Reply:
bdata is always a new anndata object in this code (with references to adata's objects). 

What's happening here is we have a new bdata object, we validate it by creating a new anndata manager (via transfer).

Then we add the new latent fields to the new manager, and finally swap out the manager and adata associated with the model (so the model is no longer associated with adata)

Reply:
I think. my question still remains. In particular, I dont understand why we are creating a new `adata` object (`bdata`), only to have it mirror `self.adata` 1:1.

Reply:
There are two options:

1. `minify_adata` is False and nothing happens w.r.t. user data expect the new fields
2. `minify_adata` is True and a truly minimal adata is attached to the model. No data is copied as it's all references to adata objects in memory.

Reply:
I get that. But my question is why create a new anndata object, call it bdata, (optionally "minify" it,) and replace adata with bdata. Why not just do everything directly on self.adata?